packages/search-adapter/src/empathy/mappers/request/params/empathy-request-related-tags-query.mapper.ts

```diff
@@ -8,9 +9,29 @@ import { RequestMapper } from '../../../empathy-adapter.types';
  * @public
  */
 @injectable()
-export class EmpathyRequestRelatedTagsQueryMapper implements RequestMapper<QueryableRequest, string> {
-
+export class EmpathyRequestRelatedTagsQueryMapper
+  implements RequestMapper<QueryableRequest, string>
+{
   map({ relatedTags = [] }: QueryableRequest, query: string): string {
-    return relatedTags.reduce((chain, rt) => `${chain} ${rt.tag}`, query).trim();
+    const [leftRelatedTags, rightRelatedTags] = this.splitRelatedTagsByQueryPosition(relatedTags);
+    return `${leftRelatedTags} ${query} ${rightRelatedTags}`.trim();
+  }
+
+  splitRelatedTagsByQueryPosition(relatedTags: RelatedTag[]): string[] {
+    return relatedTags
+      .reduce(
+        ([left, right], { tag, query }) => {
+          return this.isTagBeforeQuery(tag, query)
+            ? [`${left} ${tag}`, right]
+            : [left, `${right} ${tag}`];
+        },
+        ['', '']
+      )
+      .map(sortedRelatedTags => sortedRelatedTags.trim());
+  }
+
+  // Checks if the query is after the tag.
+  isTagBeforeQuery(tag: string, query: string): boolean {
+    return query.indexOf(tag) === 0;
```

Comment:
I just realised that you have the `startsWith` method, which is more suited for this change.